### PR TITLE
Add sibling check

### LINKED ISSUE #45: Make clear in installation docs that the sibling SDK version has to match.
We will need to document in the installation steps to ensure the version of the sibling SDK matches the version of the sentry-javascript dependencies that @sentry/capacitor depends on.

### PR DESCRIPTION
**Pull request by lucas-zimerman**
The goal is to better inform the user that Sentry/Capacitor requires the exact version on the Sibling SDKs as defined on the peerDependency.

This code will look for any @sentry/package references on the packages.json file from the user's project, if those references doesn't match the required one by the script, the script will warn the user about it when a dependency gets installed or updated.

Some questions in regard to this feature:

- Should it feature just warn the user or throw an error when installing/updating the dependency?

If we throw an error, the user will only be able to update the project if he/she updates all sibling dependencies in one command (ie yarn add @sentry/angular@7.15.0 @sentry/hub@7.15.0 @sentry/capacitor@0.11.0 --exact)

Fixes #45, #218, #268

### CODE REVIEW

File: scripts/check-siblings.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+const fs = require('fs');
+const path = require('path');
+
+//Filters all Sentry packages but Capacitor itself.
+const jsonFilter = /\s*\"\@sentry\/(?!capacitor)(?<packageName>[a-zA-Z]+)\"\:\s*\"(?<version>.+)\"/;
+const siblingVersion = "7.15.0";
```
Comment (AbhiPrasad) [inline scripts/check-siblings.js:6]: Can we just grab this from the sentry capacitor package.json?

Comment (lucas-zimerman) [reply scripts/check-siblings.js:6]: Yes we can

Comment (lucas-zimerman) [reply scripts/check-siblings.js:6]: Another option is to grab it from the environment inside of the postinstall script:
```
console.log("Core version is : "+ env.npm_package_dependencies__sentry_browser);
output
Core version is : 7.15.0
```
Not sure about the compatibility about it with npm or older yarn versions.

**Comment by lucas-zimerman** [comment]
Added more code after validating both yarn/npm.
When using yarn, we can validate the packages in two places
1) The console arguments (`yarn install @sentry/angular@7.13.0`
In this case we can simply validate if the sentry version requested is compatible with the current capacitor version (avoiding the parse of the package.json from the user).
This feature is only available on Yarn so when using NPM it'll be skipped.

2) Parsing the packages.json
This path works with both NPM and Yarn (used when user simply types `yarn install`), it'll look for any sibling reference that doesn't match the desired version or uses the `^` symbol on the sibling version.

This is a tricky code since it postinstall gets called before the package manager decides to update the packages.json .

Furthermore, this option will list all the required changes and also the required command in order to update the libraries (We could update the user's package.json but I think that's too evasive/extra effort by the script).

Ther's also a flag `--update-sentry-capacitor` to ignore the postinstall check since when updating a library using method 2, postinstall will only read the previous version prior the update, avoiding the user to be stuck on the update process).

Sample from what the user will see when installing the wrong sibling.

NPM
```
npm ERR!       throw IncompatibilityError;
npm ERR!       ^
npm ERR! This version of Sentry Capacitor is incompatible with the following installed packages:
npm ERR! @sentry/angular version ^7.13.0
npm ERR! @sentry/vue version ^7.13.0
npm ERR! Please install the mentioned packages with exactly with version 7.15.0 and with the argument --update-sentry-capacitor
npm ERR! npm install --save-exact @sentry/angular@7.15.0 @sentry/vue@7.15.0  --update-sentry-capacitor
```

Yarn
```
      throw IncompatibilityError;
      ^
This version of Sentry Capacitor is incompatible with the following installed packages:
@sentry/angular version 7.13.0
@sentry/vue version 7.13.0
Please install the mentioned packages with exactly with version 7.15.0 and with the argument --update-sentry-capacitor
yarn add --exact @sentry/angular@7.15.0 @sentry/vue@7.15.0  --update-sentry-capacitor
```

File: scripts/check-siblings.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      const capacitorPackageJson = fs.readFileSync(capacitorPackagePath, 'utf8');
+
+      const version = capacitorPackageJson.match(jsonFilter);
+      if (version && version.groups['version']) {
+            return version.groups['version'];
+      }
+      return undefined;
+}
+
+function validSentryPackageParameters(packages) {
```
Comment (kahest) [inline scripts/check-siblings.js:52]:
```suggestion
function ValidateSentryPackageParameters(packages) {
```

Comment (kahest) [reply scripts/check-siblings.js:52]: m: maybe also makes sense to add doc header to describe when exc is thrown

File: scripts/check-siblings.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+      }
+}
+if (incompatiblePackages.length > 0) {
+      let IncompatibilityError = "This version of Sentry Capacitor is incompatible with the following installed packages: \n";
+      let packagesList = ''
+      for (const package of incompatiblePackages) {
+            IncompatibilityError += package[0] + ' version ' + package[1] + '\n';
+            packagesList += package[0] + '@' + siblingVersion + ' ';
+      }
+      IncompatibilityError += 'Please install the mentioned packages with exactly with version ' + siblingVersion + ' and with the argument ' + updateArgument + '\n';
```
Comment (kahest) [inline scripts/check-siblings.js:142]:
```suggestion
      IncompatibilityError += 'Please install the mentioned packages exactly with version ' + siblingVersion + ' and with the argument ' + updateArgument + '\n';
```

File: scripts/check-siblings.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+      const version = capacitorPackageJson.match(jsonFilter);
+      if (version && version.groups['version']) {
+            return version.groups['version'];
+      }
+      return undefined;
+}
+
+function validSentryPackageParameters(packages) {
+      let errorMessages = "";
```
Comment (kahest) [inline scripts/check-siblings.js:53]: m: could collect messages in an array and join on throw

**Review by kahest — COMMENTED** [review]
Thanks for this! 🚀 
LGTM, some minor suggestions from my side.

**Review by AbhiPrasad — COMMENTED** [review]
Hey @lucas-zimerman sorry for the late review.

Want to see if we can maybe pivot on this strategy a little - can we try doing something like we do here? https://github.com/getsentry/sentry-electron/pull/584

This way we don't need a `postinstall`, we just rely on TypeScript errors.

**Comment by lucas-zimerman** [comment]
> Hey @lucas-zimerman sorry for the late review.
> 
> Want to see if we can maybe pivot on this strategy a little - can we try doing something like we do here? [getsentry/sentry-electron#584](https://github.com/getsentry/sentry-electron/pull/584)
> 
> This way we don't need a `postinstall`, we just rely on TypeScript errors.

That would be great, but here, instead of throwing an error on the SDK path, we only duplicated errors from with @sentry/types

```
Error: node_modules/@sentry/angular/node_modules/@sentry/types/types/globals.d.ts:2:11 - error TS2451: Cannot redeclare block-scoped variable '__DEBUG_BUILD__'.

2     const __DEBUG_BUILD__: boolean;
            ~~~~~~~~~~~~~~~

  node_modules/@sentry/types/types/globals.d.ts:2:11
    2     const __DEBUG_BUILD__: boolean;
                ~~~~~~~~~~~~~~~
    '__DEBUG_BUILD__' was also declared here.


Error: node_modules/@sentry/types/types/globals.d.ts:2:11 - error TS2451: Cannot redeclare block-scoped variable '__DEBUG_BUILD__'.

2     const __DEBUG_BUILD__: boolean;
            ~~~~~~~~~~~~~~~

  node_modules/@sentry/angular/node_modules/@sentry/types/types/globals.d.ts:2:11
    2     const __DEBUG_BUILD__: boolean;
                ~~~~~~~~~~~~~~~
    '__DEBUG_BUILD__' was also declared here.



[ERROR] An error occurred while running subprocess ng.

        ng.cmd run app:build exited with exit code 1.
```

File: scripts/check-siblings.js [hunk 5]
```diff
@@ -1,36 +1,145 @@
...
+const { env } = require('process');
+
+const updateArgument = '--update-sentry-capacitor';
+
+/**
+ * If user requested to ignore the post-install
+ * @return {Boolean} true if requested to skip the post-install check, false otherwise.
+ */
+function SkipPostInstall() {
+      if (env.npm_config_update_sentry_capacitor) {
```
[the remark this answers is not in the document]
Comment (lucas-zimerman) [reply scripts/check-siblings.js:12]: somehow the editor was ignoring this format, I fixed on a new commit.

**Review by kahest — APPROVED** [review]

File: scripts/check-siblings.js [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+    return env.npm_package_dependencies__sentry_browser;
+  }
+
+  let capacitorPackagePath = '';
+  if (env.npm_package_json) {
+    // NPM.
+    capacitorPackagePath = env.npm_package_json;
+  }
+  else {
+    capacitorPackagePath = __dirname + '../Package.json';
```
Comment (piotr-cz) [inline scripts/check-siblings.js:42]: I think here should be `'../package.json'`  (lowercase) as here: https://github.com/getsentry/sentry-capacitor/blob/0.11.0/package.json

Comment (lucas-zimerman) [reply scripts/check-siblings.js:42]: Thanks for spotting this! it should be fixed on the version!

**Pull request merged**